server/models/annotation.py

```diff
@@ -0,0 +1,534 @@
+        if properyName != 'type'
+    }
+    baseRectangleShapePatternProperties.update(baseShapePatternProperties)
+    rectangleShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectangle']
+                    },
+                },
+                'required': ['type'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False
+            }
+        ]
+    }
+    rectangleGridShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectanglegrid']
+                    },
+                    'widthSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                    'heightSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                },
+                'required': ['type', 'widthSubdivisions', 'heightSubdivisions'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False,
+            }
+        ]
+    }
+
+    annotationSchema = {
+        '$schema': 'http://json-schema.org/schema#',
+        'id': '/girder/plugins/large_image/models/annotation',
+        'type': 'object',
+        'properties': {
+            'name': {
+                'type': 'string',
+                # TODO: Disallow empty?
+                'minLength': 1,
+            },
+            'description': {'type': 'string'},
+            'attributes': {
+                'type': 'object',
+                'additionalProperties': True,
+                'title': 'Image Attributes',
+                'description': 'Subjective things that apply to the entire '
+                               'image.'
+            },
+            'elements': {
+                'type': 'array',
+                'items': {
+                    # Shape subtypes are mutually exclusive, so for
+                    #  efficiency. don't use 'oneOf'
+                    'anyOf': [
+                        # If we include the baseShapeSchema, then shapes that
+                        # are as-yet invented can be included.
+                        # baseShapeSchema,
+                        arrowShapeSchema,
+                        circleShapeSchema,
+                        pointShapeSchema,
+                        polylineShapeSchema,
+                        rectangleShapeSchema,
+                        rectangleGridShapeSchema,
+                    ]
```

Comment:
Width and height is better - I just think the actual foci and conversions that are less intuitive.

Sent from my iPhone

On Apr 12, 2016, at 5:59 PM, David Manthey <notifications@github.com<mailto:notifications@github.com>> wrote:


In server/models/annotation.py<https://github.com/DigitalSlideArchive/large_image/pull/50#discussion_r59402485>:

> +            'elements': {
> +                'type': 'array',
> +                'items': {
> +                    # Shape subtypes are mutually exclusive, so for
> +                    #  efficiency. don't use 'oneOf'
> +                    'anyOf': [
> +                        # If we include the baseShapeSchema, then shapes that
> +                        # are as-yet invented can be included.
> +                        # baseShapeSchema,
> +                        arrowShapeSchema,
> +                        circleShapeSchema,
> +                        pointShapeSchema,
> +                        polylineShapeSchema,
> +                        rectangleShapeSchema,
> +                        rectangleGridShapeSchema,
> +                    ]


It is pretty easy to convert between width and height and eccentricity and semi-major axis (or even semi-inter-foci distance and semi major axis). Regardless, we'll have a rotation.

I think width and height are convenient, in that they make it easy to show the circum-rectangle, and the algorithms I've used in the past to fit ellipses generally output semi-major and semi-minor axes, but if eccentricity is a more useful metric, I'm happy to use it.

-
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/DigitalSlideArchive/large_image/pull/50/files/149c5ffc59cebe59cea00b1f051dbecc2db4c549#r59402485>

________________________________

This e-mail message (including any attachments) is for the sole use of
the intended recipient(s) and may contain confidential and privileged
information. If the reader of this message is not the intended
recipient, you are hereby notified that any dissemination, distribution
or copying of this message (including any attachments) is strictly
prohibited.

If you have received this message in error, please contact
the sender by reply e-mail message and destroy all copies of the
original message (including attachments).